syntax = "proto3";

package greet;

service Greeter{

    //unary
    rpc SayHello(HelloRequest) returns (HelloReply);

    // Server streaming
    rpc ParrotSaysHello (HelloRequest) returns (stream HelloReply);

    //Client Streaming
    rpc ChattyClientSaysHello (stream HelloRequest) returns (HelloReply);

    //2way streaming
    rpc InteractingHello (stream HelloRequest) returns (stream HelloReply);

}


// The request message containing the user's name.
message HelloRequest {
    string name = 1;
    string greeting = 2;
  }
  
  // The response message containing the greetings.
  message HelloReply {
    string message = 1;
  }
  
  message DelayedReply {
      string message = 1;
      repeated HelloRequest request = 2;
  }